src/DetectPluginAssets.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace WP2Static;
+
+use RecursiveIteratorIterator;
+use RecursiveDirectoryIterator;
+
+class DetectPluginAssets {
+
+    public static function detect() {
+        $wp_site = new WPSite();
+
+        $files = array();
+
+        /*
+            We cannot rely on the plugin's location here, as it will fail for
+            a symlinked plugin directory. Our WPSite->plugins_path is more
+            reliable.
+        */
+        $plugins_path = $wp_site->plugins_path;
```

Comment:
@szepeviktor was unable to get proper plugins path when I'm within a symlinked plugin, so using WPSite's method. Could we relax the constants warnings in PHPStan or is there a better way to get the plugins path in case of symlink (`is_link`/`readlink` weren't working for me)?

Reply:
WPSite's method is OK!!

I think you don't have to care about symlinks as they can be handles transparently.

WPSite need to be reworked.
Should I work for free or for an hourly rate?

Reply:
Please read https://codex.wordpress.org/Determining_Plugin_and_Content_Directories

Reply:
How to do it

1. make the available for all other classes - I think it should be a static array
1. group them in the method for readability, see Codex
1. `grep` WP core for each and every path and URL
1. test on couple of live :) sites

Reply:
directory structures

- standard
- [wp-content lives outside ABSPATH](https://github.com/szepeviktor/debian-server-tools/blob/master/webserver/WordPress.md#semisecret-subdirectory-structure) and optionally wp-content is renamed
- document root = wp-content, ABSPATH is a directory **below** ([very weird](https://github.com/szepeviktor/debian-server-tools/blob/master/webserver/apache-conf-available/wordpress.inc.conf#L89-L103), for hiding WordPress)

Reply:
Hourly rate is good - someone has to pay for the great work you're doing with Debian ;)

The symlink should be transparent, but the WP methods for getting uploads dir didn't seem to do it for me. My local dev is with symlinks, so will be first to complain if it breaks :) 

Happy to see a refactored WPSite!

Reply:
Just WPSite left on phpstan now, will merge to get other improvements to `develop`